chore(replay): Clarify network bandwidth and show how to reduce perf impact for mobile SDKs (#12479)
getsentry/sentry-docs: docs/platforms/android/session-replay/performance-overhead.mdx

| CPU                              | 36%             | 42%                 |
| App Startup Time (Cold)          | 1533.35 ms      | 1539.55 ms          |
| Main Thread Time                 | n/a             | 20ms                |
| Network Bandwidth                | n/a             | 7 KB/s of recording |

<Alert>

Jetpack Compose view hierarchies may be slower to snapshot initially due to ART optimizations, compared to the traditional Android View System. But their performance improves as execution progresses.

> </Note>

Author: ```suggestion
</Alert>
```

Jetpack Compose view hierarchies may be slower to snapshot initially due to ART optimizations, compared to the traditional Android View System. But their performance improves as execution progresses.

</Note>

## Reducing Performance Overhead

To minimize the performance impact of the Replay SDK, consider the following steps:

### Change Replay quality

Lowering the quality of captured screenshots and videos can significantly reduce CPU, memory, and network bandwidth usage. Here's how you can do it:

```kotlin
SentryAndroid.init(context) { options ->
  // this will reduce screenshot compression to 10 and bitrate to 50kbps
  options.sessionReplay.quality = SentryReplayQuality.LOW // defaults to MEDIUM
}
```

### Disable Replay for Low-End Devices

> If the Replay SDK causes performance issues on lower-end devices, you can disable it specifically for those devices:

Reviewer: Is it worth defining the parameters of how we qualify a device as 'low-end' here? Or is there a generally known standard?

Author: yeah I can add a link to how we classify them but in general it's up to the customer to decide what do they consider a low-end 👍

> If the Replay SDK causes performance issues on lower-end devices, you can disable it specifically for those devices:

Author: ```suggestion
If the Replay SDK causes performance issues on lower-end devices (e.g., [this](https://github.com/getsentry/relay/blob/695b459e03481f7d799f07b2b901b140e5d5753d/relay-event-schema/src/protocol/device_class.rs#L21-L37) is how Sentry determines the device class), you can disable it specifically for those devices:
```

> If the Replay SDK causes performance issues on lower-end devices (e.g., [this](https://github.com/getsentry/relay/blob/695b459e03481f7d799f07b2b901b140e5d5753d/relay-event-schema/src/protocol/device_class.rs#L21-L37) is how Sentry determines the device class), you can disable it specifically for those devices:

Author: ```suggestion
If the Replay SDK causes performance issues on lower-end devices (for example, [this](https://github.com/getsentry/relay/blob/695b459e03481f7d799f07b2b901b140e5d5753d/relay-event-schema/src/protocol/device_class.rs#L21-L37) is how Sentry determines the device class), you can disable it specifically for those devices:
```
